control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -893,6 +983,43 @@ func createKubeAPIServerService(client client.Client, hcp *hyperv1.HostedControl
 	return svc, nil
 }
 
+func createKubeAPIServerServiceNodePort(client client.Client, hcp *hyperv1.HostedControlPlane, namespace string, status *InfrastructureStatus) (*corev1.Service, error) {
```

Comment:
@csrwng should we be moving all new reconcile code to match the `CreateOrUpdate` pattern of the hosted cluster controllers?

Reply:
I think so.  We don't want `create` functions; we want `reconcile` functions that can deal with the resource already existsing.

Reply:
Agree, I think we want to follow the `CreateOrUpdate` pattern uniformly. So my preview of changes coming to the control plane operator is here: https://github.com/openshift-hive/hypershift-lite/blob/main/pkg/controllers/kubeservice/reconciler.go#L374-L383
So something following a pattern like that would be awesome.

Reply:
adjusted for node port calls